### PR TITLE
Add changelog for 4.1.0

### PR DESCRIPTION
**Pull request by xing-yang**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-4.1.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+## Changes by Kind
+
+### Deprecations
+
+- VolumeSnapshot v1beta1 is deprecated and will be removed in a future release. It is recommended for users to upgrade to VolumeSnapshot CRD version v1 as soon as possible. ([#493](https://github.com/kubernetes-csi/external-snapshotter/pull/493), [@xing-yang](https://github.com/xing-yang))
+
+### API Change
+
+- Changes VolumeSnapshot API storage version from v1beta1 to v1; VolumeSnapshot v1beta1 is deprecated and will be removed in a future release. ([#493](https://github.com/kubernetes-csi/external-snapshotter/pull/493), [@xing-yang](https://github.com/xing-yang))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-4.1.md:15]: Do we also want to make a note that any previously invalid v1beta1 objects have to be deleted before upgrading to this version?

File: README.md [hunk 2]
```diff
@@ -20,13 +20,13 @@ Blog post for the GA feature can be found [here](https://kubernetes.io/blog/2020
...
-| [CSI Spec v1.2.0](https://github.com/container-storage-interface/spec/releases/tag/v1.2.0) | k8s.gcr.io/sig-storage/snapshot-validation-webhook  | 1.17     |
+| Supported CSI Versions                                                                | Container Image             | [Min K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#minimum-version) | [Recommended K8s Version](https://kubernetes-csi.github.io/docs/project-policies.html#recommended-version)
+| ------------------------------------------------------------------------------------------ | ----------------------------| --------------- | --------------- |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0), [CSI Spec v1.1.0](https://github.com/container-storage-interface/spec/releases/tag/v1.1.0), [CSI Spec v1.2.0](https://github.com/container-storage-interface/spec/releases/tag/v1.2.0), [CSI Spec v1.3.0](https://github.com/container-storage-interface/spec/releases/tag/v1.3.0), [CSI Spec v1.4.0](https://github.com/container-storage-interface/spec/releases/tag/v1.4.0) | k8s.gcr.io/sig-storage/csi-snapshotter | 1.20         | 1.20         |
```
Comment (msau42) [inline README.md:25]: Do we need to mention all of the version? I think just mentioning 1.0 is min, and 1.4 as recommended is sufficient

File: README.md [hunk 3]
```diff
@@ -186,7 +186,9 @@ If you have already deployed v1alpha1 snapshot APIs and external-snapshotter sid
 
 Validation webhook should be installed before upgrading to v1. Potential impacts of not installing the validation webhook before upgrading to v1 include being unable to delete invalid snapshot objects. See the section on Validation Webhook for details.
 
-Change from v1beta1 to v1 is backward compatible. Both v1 and v1beta1 are served while the stored API version is still v1beta1. Future releases will switch the stored version to v1 and gradually remove v1beta1 support.
+* When upgrading to 4.0, change from v1beta1 to v1 is backward compatible because both v1 and v1beta1 are served while the stored API version is still v1beta1. Future releases will switch the stored version to v1 and gradually remove v1beta1 support.
+* When upgrading from 3.x to 4.1, change from v1beta1 to v1 is no longer backward compatible because stored API version is changed to v1 although both v1 and v1beta1 are still served. V1beta1 is deprecated in 4.1.
```
Comment (msau42) [inline README.md:190]: nit: lowercase v1beta1

File: CHANGELOG/CHANGELOG-4.1.md (hunk 1 above)
Comment (xing-yang) [reply CHANGELOG/CHANGELOG-4.1.md:15]: Sure.  Added in Deprecations section.

File: README.md (hunk 2 above)
Comment (xing-yang) [reply README.md:25]: Sure.

File: README.md (hunk 3 above)
Comment (xing-yang) [reply README.md]: Done

**Pull request merged**